src/components/ec-inline-input-field/components/edit/edit.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <div
+    class="tw-w-full tw-flex"
+    data-test="ec-inline-input-field-edit"
+  >
+    <ec-input-field
+      ref="input"
+      v-model="value"
+      type="text"
+      data-test="ec-inline-input-field-edit__input"
+      @keydown.enter="submitViaKeyboard"
+      @keydown.esc="cancelViaKeyboard"
+    />
+    <div class="tw-flex tw-ml-20 tw-mt-12">
+      <button
+        class="ec-inline-input-field-edit__action"
+        data-test="ec-inline-input-field-edit__submit-action"
+        @click="submit"
+        @keydown.enter.space.prevent="submitViaKeyboard"
+      >
+        <ec-icon
+          class="ec-inline-input-field-edit__action-icon"
+          name="simple-check"
+          :size="16"
+        />
+      </button>
+      <button
+        class="ec-inline-input-field-edit__action tw-ml-8"
+        data-test="ec-inline-input-field-edit__cancel-action"
+        @click="cancel"
+        @keydown.enter.space.prevent="cancelViaKeyboard"
+      >
+        <ec-icon
+          class="ec-inline-input-field-edit__action-icon"
+          name="simple-close"
+          :size="16"
+        />
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import { EDITING, LOADING, READ_ONLY } from '@/enums/input-status';
+import EcIcon from '@/components/ec-icon';
+import EcInputField from '@/components/ec-input-field';
+
+export default {
+  name: 'EcInlineInputFieldEdit',
+  components: { EcIcon, EcInputField },
+  props: {
+    originalValue: {
+      default: '',
+      type: String,
+    },
+    status: {
+      type: String,
+      default: READ_ONLY,
+      validator(value) {
+        return [READ_ONLY, EDITING, LOADING].includes(value);
+      },
+    },
+  },
+  data() {
+    return {
+      value: null,
+    };
+  },
+  watch: {
+    status: {
+      immediate: true,
+      handler() {
+        this.value = this.originalValue;
+        this.$nextTick(() => {
+          this.$refs.input.$el.querySelector('input').focus();
```

Comment:
I'm not very big fan of parent being aware of what's in the children. I know it was used like this somewhere in ec-dropdown, and I see how quickly bad design spreads. There is not much to do here, just make it a little better, we can create a method in ec-input-field:
```
methods: {
  focus() {
    this.$refs.input.focus();
  }
}
```
and then here you have that function exposes and you can use it in this component as `this.$refs.input.focus()`. I'm not still satisified with this solution but at least we got rid of `querySelector('input')`.